test/all-templates.test.js

```diff
@@ -197,6 +212,97 @@ describe('CI template verification', () => {
           done();
         });
       });
+
+      it('should be a subset of the repo package.json', (done) => {
+        const repoPackageJson = path.join(projectRoot, 'package.json');
+        const missingDeps = [];
+
+        fs.readFile(repoPackageJson, (err, contents) => {
+          expect(err).toBeFalsy();
+          const repoPackageData = JSON.parse(contents);
+
+          fs.readFile(packageJson, (err, contents) => {
+            expect(err).toBeFalsy();
+            const appPackageData = JSON.parse(contents);
+
+            for (const dep of Object.keys(appPackageData.dependencies)) {
+              if (!repoPackageData.devDependencies[dep]) {
+                missingDeps.push(dep);
+              }
+            }
+
+            if (missingDeps.length > 0) {
+              throw new Error(
+                `The repo package.json is missing these dependencies: ${missingDeps}. Did you use 'npm run add-dependency'?`
+              );
+            }
+
+            done();
+          });
+        });
+      });
+    });
+
+    describe('its templates.json entry', () => {
+      it('should exist and have all required fields', () => {
+        if (!templatesMap[template]) {
+          throw new Error(`${template} does not have a templates.json entry`);
+        }
+        const entry = templatesMap[template];
+
+        if (!entry.name) {
+          throw new Error(
+            `${template} does not have a "name" field in templates.json`
+          );
+        } else if (!entry.description) {
+          throw new Error(
+            `${template} does not have a "description" field in templates.json`
+          );
+        }
+      });
+    });
+
+    describe('its unit tests', () => {
+      it('should have one per Function', (done) => {
+        const functionsDir = path.join(projectRoot, template, 'functions');
+        const testsDir = path.join(projectRoot, template, 'tests');
+        const missingTests = [];
+
+        if (incompleteTests.includes(template)) {
+          done();
+          return;
+        }
+
+        fs.readdir(functionsDir, (err, functions) => {
```

Comment:
This only tests the top level functions in the `functions` directory and `tests` directory. Some functions have a directory structure within the `functions` directory, like the [voice-client-javascript](https://github.com/twilio-labs/function-templates/tree/main/voice-client-javascript). This means you can avoid this test by entering your function in the `incompleteTests` array or by placing all your functions in a directory inside the `functions` directory.

Reply:
Yep! This is expected behavior for the time being to keep the verification simple; I have documented the behavior though, since it's probably rather surprising otherwise.